testing/scenario_app/lib/src/platform_view.dart

```diff
@@ -366,6 +366,75 @@ class PlatformViewMaxOverlaysScenario extends Scenario
   }
 }
 
+/// A platform view with adjacent surrounding layers should not create overlays.
+class PlatformViewSurroundingLayersScenario extends Scenario
+    with _BasePlatformViewScenarioMixin {
+  /// Creates the PlatformView scenario.
+  PlatformViewSurroundingLayersScenario(
+    super.view, {
+    required this.id,
+  });
+
+  /// The platform view identifier.
+  final int id;
+
+  @override
+  void onBeginFrame(Duration duration) {
+    final SceneBuilder builder = SceneBuilder();
+
+    // Simulate partial pixel offsets as we would see while scrolling.
+    // All objects in the scene below are then on sub-pixel boundaries.
+    builder.pushOffset(0.5, 0.5);
+
+    // a platform view from (100, 100) to (200, 200)
+    builder.pushOffset(100, 100);
+    addPlatformView(
+      id,
+      width: 100,
+      height: 100,
+      dispatcher: view.platformDispatcher,
+      sceneBuilder: builder,
+    );
```

Comment:
You need a pop here to take off the 100x100 offset otherwise the drawRect calls will overlap the PV fully.

Reply:
Ah, I just realized that you pop this 100x100 offset below before you call addPicture but the logical nesting "appears" off to a casual code read. And in the end, you never do pop the 0.5 offset (which is probably handled by the "build" method anyway, but it would be nice if the nested offset layers are explicitly popped at semantically appropriate points during the build process).

So, I think the test case is correct, but doesn't read very obviously...

Reply:
updated `pop()`